core/scripts.py

```diff
@@ -3420,9 +3420,21 @@ def should_process_lane(self, lane):
 
     def process_lane(self, lane):
         """Update the estimated size of a Lane."""
+
+        # We supress the configuration changes updates, as each lane is updated
+        # and call the site_configuration_has_changed function once after this
+        # script has finished running.
+        #
+        # This is done because calling site_configuration_has_changed repeatedly
+        # was causing performance problems, when we have lots of lanes to update.
+        lane._suppress_configuration_changes = True
```

Comment:
Is this ever set back to `False` afterwards? Maybe I'm missing it.

Reply:
I'd be inclined to wrap `update_size()` in a `try ... finally` and set the flag back to `False` in the `finally`.

Reply:
My thought was, that since `_suppress_configuration_changes` is set at on the instance, and not persisted to the database, that there is no need to set back to false, since any new loads will give new lane objects with the default of `False`, and the lifetime of these `Lane` objects is just within the script, where we want the config changes suppressed. 

I pushed another commit to make this a bit more explicit, but defining `_suppress_configuration_changes` on the instance, inside the `__init__` instead of on the class. Previously it was defined on the class, but only set at the instance level.

Reply:
Yeah, my comment only applies to the class variable version. :slightly_smiling_face: